packages/enhanced/src/lib/container/runtime/CustomRuntimePlugin.ts

```diff
@@ -0,0 +1,240 @@
+import { normalizeWebpackPath } from '@module-federation/sdk/normalize-webpack-path';
+import CustomRuntimeModule from './CustomRuntimeModule';
+const { RuntimeGlobals } = require(
+  normalizeWebpackPath('webpack'),
+) as typeof import('webpack');
+import type { Compiler, Compilation, Chunk, Module, ChunkGraph } from 'webpack';
+import { getFederationGlobalScope } from './utils';
+import fs from 'fs';
+import path from 'path';
+
+const onceForCompilationMap = new WeakMap();
+const federationGlobal = getFederationGlobalScope(RuntimeGlobals);
+import { ConcatSource } from 'webpack-sources';
+
+class RuntimeModuleChunkPlugin {
+  apply(compiler: Compiler): void {
+    compiler.hooks.thisCompilation.tap(
+      'ModuleChunkFormatPlugin',
+      (compilation: Compilation) => {
+        compilation.hooks.optimizeModuleIds.tap(
+          'ModuleChunkFormatPlugin',
+          (modules: Iterable<Module>) => {
+            for (const module of modules) {
+              const moduleId = compilation.chunkGraph.getModuleId(module);
+              if (typeof moduleId === 'string') {
+                compilation.chunkGraph.setModuleId(
+                  module,
+                  `(embed)${moduleId}`,
+                );
+              } else {
+                compilation.chunkGraph.setModuleId(module, `1000${moduleId}`);
```

Comment:
create module that should not conflict with other module in tree, needs improvement